### PR TITLE
Site Logo: Core compat

### LINKED ISSUE #3447: Site Logo: ensure compatibility with Core Site Logo feature
Jetpack's [Site Logo](http://jetpack.me/support/site-logo) theme tool might be added to Core in WordPress 4.5.:
[#33755-core](https://core.trac.wordpress.org/ticket/33755)

If that happens, we'll need to make sure we handle fallback nicely, like we did for Site Icon.

### PR DESCRIPTION
**Pull request by dereksmart**
#Fixes #3447

This acts as a Minimal Viable Product to make Jetpack's Site Logo code compatible with Core.  It also makes it so that themes currently using Jetpack's Site Logo will "work" with core's logo.  

Basically all we're doing here is transferring the site_logo ID to the theme_mod() that core uses, and if it's there we use that in the `jetpack_the_site_logo()` function that themes use to get the logo.  We also kill all the customizer/settings hooks that Jetpack registers, in favor of core's setting section.  

It's kinda weird in that we're using core's settings code, but our display code right now.  BUT this is totally up to the theme authors to change their themes to use core.  

cc @obenland

### CODE REVIEW

**Comment by obenland** [comment]
Core's version will be renamed to `custom-logo`, as decided in today's dev chat.
Compat will likely be passing on theme support in addition to what you have here. Not sure yet.

File: modules/theme-tools/site-logo/inc/functions.php [hunk 1]
```diff
@@ -102,6 +102,7 @@ function jetpack_has_site_logo() {
  */
 function jetpack_the_site_logo() {
 	$logo = site_logo()->logo;
+	$logo_id = get_theme_mod( 'site_logo' ) ?: $logo['id'];
```
Comment (kraftbj) [inline modules/theme-tools/site-logo/inc/functions.php]: `?:` is a 5.3+ operator.

**Comment by zinigor** [comment]
Thanks! Tested with a theme that uses Jetpack's site logo, everything seems to work fine after upgrading to 4.5, no theme errors. Let's get this in like that, and then adapt to further changes if needed.

**Comment by obenland** [comment]
I'd recommend testing again now, after https://core.trac.wordpress.org/changeset/36837 and https://core.trac.wordpress.org/changeset/36838 made it in.

**Comment by dereksmart** [comment]
It's been updated to take the name changes into account and refactors a little bit so that and jetpack logo will be automatically transferred to core's if there is one.

**Comment by zinigor** [comment]
Cool, thank you!
* It works using Jetpack until the theme announces support for the new slug;
* Gracefully turns the control over to Core when needed.
Look good to go!

**Pull request merged**